### ISSUE TITLE
Conda install of [neuralforecast] is broken

### ISSUE THREAD

**Issue by twobitunicorn** [Nixtla/neuralforecast#834]
### What happened + What you expected to happen

Doing a straight up install on a python=3.11.* conda environment gives the following error.  This is from executing the code `conda install -c conda-forge neuralforecast`.

Same ticket just a bit older.  https://github.com/Nixtla/neuralforecast/issues/440

```Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): - WARNING conda.models.version:get_matcher(546): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.8.0.*, but conda is ignoring the .* and treating it as 1.8.0
WARNING conda.models.version:get_matcher(546): Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda. Your spec was 1.9.0.*, but conda is ignoring the .* and treating it as 1.9.0
done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versionsThe following specifications were found to be incompatible with your system:

  - feature:/osx-arm64::__osx==14.0=0
  - neuralforecast -> numba -> __osx[version='>=10.9|>=11.0']
  - python=3.11.6 -> ncurses[version='>=6.4,<7.0a0'] -> __osx[version='>=10.9']

Your installed version is: 14.0

Note that strict channel priority may have removed packages required for satisfiability.c
```

### Versions / Dependencies

conda

### Reproduction script

conda create --name nixtla python=3.11.6
conda install -c conda-forge neuralforecast

### Issue Severity

Low: It annoys or frustrates me.

**Comment by jmoralez** [comment]
Hey. Can you try the following:
* Merging the installs: `conda create -n nixtla -c conda-forge python=3.11.6 neuralforecast`
* Adding other channels: `conda create -n nixtla -c conda-forge -c pytorch -c defaults python=3.11.6 neuralforecast`
* Using mamba: `conda install -n base -y mamba && mamba create -n nixtla -c conda-forge python=3.11.6 neuralforecast`

**Comment by twobitunicorn** [comment]
@jmoralez Yes.  Ran ran them on my macbook pro. They each throw the same error.  See attached. 
If I remove `neuralforecast` and install only `python=3.11.6` (and/or `pytorch`) and then run `pip install neuralforecast` it works just fine.

[out.log](https://github.com/Nixtla/neuralforecast/files/13530923/out.log)

**Comment by jmoralez** [comment]
Are you able to install just numba?

**Comment by twobitunicorn** [comment]
Yes.  Running `conda create -n nix -c conda-forge python=3.11.6 numba` installs `namba=0.58.1` no problems.

**Comment by jmoralez** [comment]
And installing neuralforecast after?

**Comment by twobitunicorn** [comment]
Same issue.  I have been looking over the feedstock and it seems that the script for `osx` was taken out `.scripts/run_osx_build.sh`.  This is still referenced in `build-locally.py`.  The verify config only seems to support linux now as that is all it gets from the ci support directory `.ci_support/*.yaml`.

**Comment by jmoralez** [comment]
The file is noarch, so it has to be a problem in one of the dependencies.

**Comment by jmoralez** [comment]
Neither of us has an arm64 mac, so we can't debug. We appreciate you trying things out.

**Comment by twobitunicorn** [comment]
your welcome, appreciate all the work you all have done!   

It is the same issue.  Attached the output.  I have been looking over the feedstock and I think it isn't supporting the os x platform as the scripts for OSX have been removed.  It also seems that the majority of linux platforms are also currently not supported.  I can take a crack at this over the weekend.

**Comment by twobitunicorn** [comment]
I played with `--no-deps` and `--only-deps`.  Same issue with both of those.

For funsies I went through each of the deps (`environment.yml`) and did an individual conda/pip install on each one and had no issues.

**Comment by twobitunicorn** [comment]
Jumped on my other machine.  Running `mamba install -c conda-forge neuralforecast --only-deps` gives the following error below.    I can install `pyppmd` with either conda or pip.  `ray-tune` is constrained by `<3.11.0a0` on conda/mamba but `pip install ray[tune]` installs no problem.  Executing `pip install neuralforecast` (with or without a pip install of `pyppmd` and `ray-tune`)  installs just fine. 

```
Could not solve for environment specs
The following packages are incompatible
└─ neuralforecast   is installable with the potential options
   ├─ neuralforecast [0.0.7|0.0.8|...|1.6.0] would require
   │  └─ pyppmd >=0.17.0,<0.18.0 , which does not exist (perhaps a missing channel);
   └─ neuralforecast [1.6.0|1.6.1|1.6.2|1.6.3|1.6.4] would require
      └─ ray-tune >=2.2.0  with the potential options
         ├─ ray-tune 2.3.0 would require
         │  └─ python >=3.10,<3.11.0a0 , which can be installed;
         ├─ ray-tune 2.3.0 would require
         │  └─ python >=3.8,<3.9.0a0 , which can be installed;
         └─ ray-tune 2.3.0 would require
            └─ python >=3.9,<3.10.0a0 , which can be installed.
```

**Comment by jmoralez** [comment]
> For funsies I went through each of the deps (environment.yml) and did an individual conda/pip install on each one and had no issues.

Even ray-tune? It doesn't seem like they publish the conda package for arm64 ([files](https://anaconda.org/conda-forge/ray-tune/files)).

**Comment by twobitunicorn** [comment]
Apologies.  That was the one I had to do the pip in conda/pip.  Same with pyppmd.

**Comment by jmoralez** [comment]
After you install ray-tune from pip are you able to install neuralforecast from conda?

**Comment by twobitunicorn** [comment]
No.  It doesn't seem to pick up the pip package.
